spec/models/ldap_identity_provider_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe LdapIdentityProvider, type: :model do
+
+  subject { auth_provider.identity_provider }
+  let(:auth_provider) { FactoryGirl.create(:openid_authentication_service, :with_ldap_identity_provider) }
+  let(:test_user) { FactoryGirl.attributes_for(:user) }
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of :ldap_base }
+  end
+
+  describe 'affiliate' do
+    it { is_expected.to respond_to(:affiliate) }
```

Comment:
https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/respond-to-matcher